src/Microsoft.Framework.PackageManager/Restore/NuGet/HttpSource.cs

```diff
@@ -129,7 +129,7 @@ internal class HttpSource
             {
                 HttpStatusCode statusCode;
 
-                using (var response = await _client.SendAsync(request))
+                using (var response = await _client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead))
                 {
                     response.EnsureSuccessStatusCode();
```

Comment:
if I am not wrong, following should work without the need for deep nesting of `using`:
1. make request with responseheadersread, ensure success status code...AFAIK we need not dispose the response here as its the response stream in step 3. that needs to be disposed.
2. create a file stream
3. using(response's stream) { copy to the file stream }

@Tratcher is the expert here so he can validate :smile: